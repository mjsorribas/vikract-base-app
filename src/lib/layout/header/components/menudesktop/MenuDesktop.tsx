import { Box, Flex, Spacer, Link, Stack } from "@chakra-ui/react";

import MenuAccess from "../menuaccess/menuAccess";
import Logo from "lib/components/logo/Logo";
import { theme } from "lib/styles/customTheme";

export default function MenuDesktop() {
  return (
    <Flex>
      <Logo boxSize="25px" fontSize="1.168rem" fontColor={theme.colors.black} />
      <Spacer />
      <Box display="flex" alignItems="center">
        <Stack direction="row" spacing={8}>
          <Link
            fontWeight="bold"
            fontSize={{ sm: "0.875rem", md: "1.168rem" }}
            ml={5}
            _hover={{ textDecoration: "none" }}
            href="/"
          >
            Home
          </Link>
          <Link
            ml={5}
            fontSize={{ sm: "0.875rem", md: "1.168rem" }}
            _hover={{ textDecoration: "none" }}
            href="/aboutus"
          >
            About Us
          </Link>
          <Link
            ml={5}
            fontSize={{ sm: "0.875rem", md: "1.168rem" }}
            _hover={{ textDecoration: "none" }}
            href="/prices"
          >
            Prices
          </Link>
          <Link
            ml={5}
            fontSize={{ sm: "0.875rem", md: "1.168rem" }}
            _hover={{ textDecoration: "none" }}
            href="/contactus"
          >
            Contact Us
          </Link>
        </Stack>
      </Box>
      <Spacer />
      <MenuAccess />
    </Flex>
  );
}
