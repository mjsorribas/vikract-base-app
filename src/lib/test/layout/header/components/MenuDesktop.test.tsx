import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { ChakraProvider } from "@chakra-ui/react";
import { expect, test, vi } from "vitest";

// Mock de una dependencia específica
vi.mock("@chakra-ui/react", async () => {
  const actual = await vi.importActual("@chakra-ui/react");
  return {
    ...actual,
    useBreakpointValue: () => "mockedValue",
    extendTheme: actual.extendTheme, // Asegúrate de incluir extendTheme en el mock
  };
});

import MenuDesktop from "lib/layout/header/components/menudesktop/MenuDesktop";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("MenuDesktop", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <ChakraProvider>
        <MenuDesktop />
      </ChakraProvider>
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
