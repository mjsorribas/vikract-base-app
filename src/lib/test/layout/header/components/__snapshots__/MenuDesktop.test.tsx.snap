// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuMobile 1`] = `
<div
  className="css-k008qs"
>
  <div
    className="css-70qvj9"
  >
    <div
      className="css-0"
    >
      <img
        className="chakra-image css-1chmwhm"
        src="/assets/vikract.svg"
        title="The Vikract Webapp template"
      />
      <p
        className="chakra-text css-104caxi"
      >
        THE VIKRACT TEMPLATE
      </p>
    </div>
  </div>
  <div
    className="css-17xejub"
  />
  <div
    className="css-70qvj9"
  >
    <div
      className="chakra-stack css-5ax1kt"
    >
      <a
        className="chakra-link css-177s7ri"
        href="/"
      >
        Home
      </a>
      <a
        className="chakra-link css-1jpe0vt"
        href="/aboutus"
      >
        About Us
      </a>
      <a
        className="chakra-link css-1jpe0vt"
        href="/contactus"
      >
        Contact Us
      </a>
    </div>
  </div>
  <div
    className="css-17xejub"
  />
  <div
    className="css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-stack css-1u5xrkq"
      >
        <button
          className="chakra-button css-4xx2wk"
          type="button"
        >
          Sign Up
        </button>
        <button
          className="chakra-button css-4xx2wk"
          type="button"
        >
          Sign In
        </button>
      </div>
    </div>
  </div>
</div>
`;
